### PR TITLE
Add the main MAAS nav

### PR DESCRIPTION
**Pull request by huwshimi**
Done:
- Fetch the authenticated user details over the websocket. Block loading the app on getting that user.
- Add the main MAAS nav, same as the existing one. Fixes: https://github.com/canonical-web-and-design/maas-squad/issues/1337. 

QA:
- Make sure your `config.js` has all the fields from `config.example.js`.
- `./run`
- See that the header is the same as MAAS.
- Clicking on anything but Settings should take you to the MAAS set in your config.js.

### CODE REVIEW

File: ui/src/app/base/components/Header/Header.js [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+      role="menuitem"
+    >
+      <Link to={url} className="p-dropdown__item">
+        {label}
+      </Link>
+    </li>
+  );
+
+  return (
+    <header className="p-navigation">
```
Comment (huwshimi) [inline ui/src/app/base/components/Header/Header.js]: This is all largely copied from MAAS with a few React updates.

File: ui/src/scss/_patterns_navigation.scss [hunk 2]
```diff
@@ -0,0 +1,112 @@
+//Local overrides to the navigation pattern
+@mixin maas-navigation {
```
Comment (huwshimi) [inline ui/src/scss/_patterns_navigation.scss]: Copied from MAAS.

File: ui/src/scss/_settings.scss [hunk 3]
```diff
@@ -1,3 +1,11 @@
...
+$grid-max-width: 1440 / 16 * 1rem; // express in rems for easier calculations
+
+$color-navigation-background: #333;
+$color-navigation-background--hover: darken($color-navigation-background, 3%);
+
 $color-brand: #e95420;
+$color-navigation-active-bar: $color-brand;
+
+$breakpoint-navigation-threshold: 870px;
+$increase-font-size-on-larger-screens: false;
```
Comment (huwshimi) [inline ui/src/scss/_settings.scss]: This is all copied from MAAS (required for the header).

File: ui/src/app/base/components/Header/Header.js [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+    evt.preventDefault();
+    toggleMobileMenu(!mobileMenuVisible);
+  };
+
+  const _toggleHardware = evt => {
+    evt.preventDefault();
+    toggleHardware(!hardwareVisible);
+  };
+
+  const _generateURL = url => `${MAAS_config.ui.maas_url}/${url}`; // eslint-disable-line no-undef
```
[the remark this answers is not in the document]
Comment (huwshimi) [reply ui/src/app/base/components/Header/Header.js]: This sends them out of the React app and over to MAAS. In dev at least these will be on different domains.

[the remark this answers is not in the document]
Comment (anthonydillon) [reply ui/src/app/base/components/Header/Header.js]: Ah of course, forgot this wasn't the full app yet.

File: ui/src/app/base/components/Header/Header.js [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+import ReactRouterPropTypes from "react-router-prop-types";
+
+import "./Header.scss";
+import { AuthUserShape } from "../../proptypes";
+
+export const Header = ({ authUser, location }) => {
+  const [hardwareVisible, toggleHardware] = useState(false);
+  const [mobileMenuVisible, toggleMobileMenu] = useState(false);
+
+  const _toggleMobileMenu = evt => {
```
Comment (squidsoup) [inline ui/src/app/base/components/Header/Header.js]: These toggles and event handlers could be cleaned up nicely with a `useToggle` hook. David Ceddia has written a post about exactly that here: https://daveceddia.com/custom-hooks/ 

What do you think?

**Pull request merged**